tsfc/driver.py

```diff
@@ -179,6 +179,8 @@ def facetarea():
         params = {}
         # Record per-integral parameters
         params.update(integral.metadata())
+        if params.get("quadrature_rule") == "default":
```

Comment:
My reasoning for being able to provide a `QuadratureRule` object is that a third-party user can directly provide the quadrature without needing to go in and add some information to a mapping table.